h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostNativeMojoModel.java

```diff
@@ -0,0 +1,171 @@
+package hex.genmodel.algos.xgboost;
+
+import hex.genmodel.GenModel;
+import hex.genmodel.MojoModel;
+import ml.dmlc.xgboost4j.java.*;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Please note: user is advised to explicitly release the native resources of XGBoost by calling close method on the instance.
+ */
+public final class XGBoostNativeMojoModel extends XGBoostMojoModel {
+  Booster _booster;
+
+  public XGBoostNativeMojoModel(String[] columns, String[][] domains, String responseColumn) {
+    super(columns, domains, responseColumn);
+  }
+
+  public final double[] score0(double[] doubles, double offset, double[] preds) {
+    return score0(doubles, offset, preds,
+            _booster, _nums, _cats, _catOffsets, _useAllFactorLevels,
+            _nclasses, _priorClassDistrib, _defaultThreshold, _sparse);
+  }
+
+  public static double[][] bulkScore0(double[][] doubles, double[] offsets, double[][] preds,
+                                      Booster _booster, int _nums, int _cats,
+                                      int[] _catOffsets, boolean _useAllFactorLevels,
+                                      int nclasses, double[] _priorClassDistrib,
+                                      double _defaultThreshold, boolean _sparse) {
+    if (offsets != null) throw new UnsupportedOperationException("Unsupported: offset != null or only 0s");
+    float[][] floats;
+    int cats = _catOffsets == null ? 0 : _catOffsets[_cats];
+    // convert dense doubles to expanded floats
+    floats = new float[doubles.length][_nums + cats]; //TODO: use thread-local storage
+    for(int i = 0; i < doubles.length; i++) {
+      GenModel.setInput(doubles[i], floats[i], _nums, _cats, _catOffsets, null, null, _useAllFactorLevels, _sparse /*replace NA with 0*/);
+    }
+    float[][] out = null;
+    DMatrix dmat = null;
+    try {
+      Map<String, String> rabitEnv = new HashMap<>();
+      rabitEnv.put("DMLC_TASK_ID", "0");
+      Rabit.init(rabitEnv);
+      dmat = new DMatrix(floats,doubles.length,floats[0].length, _sparse ? 0 : Float.NaN);
+//      dmat.setWeight(new float[]{(float)weight});
+      out = _booster.predict(dmat);
+      Rabit.shutdown();
+    } catch (XGBoostError xgBoostError) {
+      throw new IllegalStateException("Failed XGBoost prediction.", xgBoostError);
+    } finally {
+      BoosterHelper.dispose(dmat);
+    }
+
+    for(int r = 0; r < out.length; r++) {
+      if (nclasses > 2) {
+        for (int i = 0; i < out[0].length; ++i)
+          preds[r][1 + i] = out[r][i];
+//      if (_balanceClasses)
```

Comment:
Why is this disabled ?

Reply:
it was never supported, I will delete the code, it is confusing